pkg/artifactory/datasource/datasource_artifactory_group.go

```diff
@@ -0,0 +1,62 @@
+package datasource
+
+import (
+	"fmt"
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/security"
+	"github.com/jfrog/terraform-provider-shared/packer"
+	"github.com/jfrog/terraform-provider-shared/predicate"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"golang.org/x/net/context"
+	"net/http"
+)
+
+func ArtifactoryGroupSchema() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: DataSourceGroupRead,
+		Schema:      getDataSourceGroupSchema(),
+		Description: "Provides the Artifactory Group data source. Contains information about the group configuration and optionally provides its associated user list.",
+	}
+}
+
+func getDataSourceGroupSchema() map[string]*schema.Schema {
```

Comment:
These funcs should be inside `ArtifactoryGroupSchema()` to limit their scope.

Reply:
by these funcs you mean getDataSourceGroupSchema and DataSourceGroupRead?

Reply:
Yep